docs2/site/docs/getting-started/custom-scalars.md

```diff
@@ -183,105 +213,303 @@ public class ExampleSchema : Schema
 
+```graphql
+{
+    widget (id: "3") {
+        name
+    }
+}
 ```
 
-This is necessary since the query executor converts all arguments and variables to `IValue` instances before coercing them to their server-side representation using `ParseLiteral`. `Vector3` instances parsed from variables will be converted to the more specific `Vector3Value` type.
+However, your code may use integer identifiers in the data access layer. So when you call `context.GetArgument<int>("id")`,
+GraphQL.NET calls the value converter to convert the string to an integer.
 
-7. Implement `ScalarGraphType.Serialize` so `Vector3` instances can be sent to the client.
+The value converter can be extended globally by calling the static method `Register` as follows:
 
 ```csharp
-// In Vector3Type
+ValueConverter.Register<Vector3, string>(v => $"{v.X},{v.Y},{v.Z}");
+```
 
-public override object Serialize(object value)
+The above method registers a conversion from the `Vector3` format to a `string`. Since the registration is static,
+it should only be done once per application lifetime. For instance, in a static constructor of your schema.
+
+```csharp
+public class MySchema : Schema
 {
-    return ValueConverter.ConvertTo(value, typeof(Vector3));
+    static MySchema()
+    {
+        ValueConverter.Register<Vector3, string>(v => $"{v.X},{v.Y},{v.Z}");
+    }
+
+    ...
 }
 ```
 
-This implementation may surprise you. Why is `Serialize`, which is used for output, implemented identically to `ParseValue`, which is used for input? Why does `Serialize` return an object, rather than a string or byte array? It helps to understand a few internals of the library.
+## Null values
 
-- `Serialize` will be called during query execution, and should be passed an instance of `Vector3` from a field resolver.
+Custom scalars process and handle null values during serialization and deserialization. This allows
+for custom scalars that can assist when you have database values such as 0 that should represent null
+when exposed outside the schema. Below is an example of a scalar intended to represent a database
+autoincrementing numeric identifier internally, where null values are stored as 0.
 
-- `Serialize` is _also_ called when reading variables from the client so that variables can be converted to `IValue` instances. In the case of `Vector3Type`, `value` will be a string during this process.
+```csharp
+public class DbIdGraphType : ScalarGraphType
+{
+    public DbIdGraphType()
+    {
+        Name = "DbId";
+    }
 
-- `ValueConverter.ConvertTo` handles the case when `value` is an instance of the requested type by returning `value`. Therefore, `ValueConverter.ConvertTo` neatly handles both input and output representations of the scalar.
+    public override object ParseLiteral(IValue value) => value switch
+    {
+        StringValue s => int.TryParse(s.Value, out int i) && i > 0 ? i : throw new FormatException($"'{s.Value}' is not a valid identifier."),
+        null => 0,
```

Comment:
`null` or `NullValue` or both ?

Reply:
```suggestion
        NullValue _ => 0,
```
You're right; `NullValue` is proper.  `value` cannot be `null` here.